### PR TITLE
FIx: Fixed issue where the barrier overlaps with the ship

### PR DESCRIPTION
**Pull request by bamcasa**
This pull request includes several changes aimed at improving the game's initialization and item management. The most significant changes involve fixing a height calculation bug, modifying the `ItemManager` constructor to accept dynamic screen dimensions, and adding logging for better debugging.

### Bug Fixes

* Fixed height calculation in the `Frame` constructor to correctly subtract both the top and bottom insets. (`src/engine/Frame.java`)

### Feature Enhancements

* Modified the `ItemManager` constructor to accept dynamic height and width parameters, allowing for more flexible screen dimension handling. (`src/engine/ItemManager.java`)
* Added logging to track the creation of barriers and the player ship, which helps in debugging and understanding the game's state during initialization. (`src/engine/ItemManager.java`, `src/screen/GameScreen.java`) [[1]](diffhunk://#diff-a48f84a62f34e8f90c5c1a0e1606c8eae7b6e7fcbeef4c31ce1094ef024677acL245-R254) [[2]](diffhunk://#diff-2cb83b1e334bad77df1b4983953635325b87a363c4518f6d3e9679e48be0b648L187-R195)

### Refactoring

* Changed `WIDTH` and `HEIGHT` from `static final` to `static` in `ItemManager` to allow dynamic assignment. (`src/engine/ItemManager.java`)
* Updated the `GameScreen` initialization to pass dynamic dimensions to the `ItemManager`. (`src/screen/GameScreen.java`)

### Minor Changes

* Updated binary data in `res/graphics` to reflect new graphical changes. (`res/graphics`)

### CODE REVIEW

File: res/graphics [hunk 1]
```diff
@@ -46,8 +46,7 @@
 1110111111011111101110000000111011111101111110111
 
 # Web
-100000000001001111111100001100001100001010010100001001100100001010010100001111111100100000000001
-
+100000010000000001111110011000100101011001001010010010100101011001100010011111100000000010000001
```
Comment (dev-moonsc) [inline res/graphics:49]: The graphics of the web entity have been confirmed by the Level Design team to be as originally intended. Please revert them back to their original state.

File: src/engine/ItemManager.java [hunk 2]
```diff
@@ -27,9 +27,9 @@
  */
 public class ItemManager {
     /** Width of game screen. */
-    private static final int WIDTH = 600;
+    private static int WIDTH = 600;
     /** Height of game screen. */
-    private static final int HEIGHT = 650;
+    private static int HEIGHT = 650;
```
Comment (dev-moonsc) [inline src/engine/ItemManager.java:32]: Field variable WIDTH and Height are initialized when the itemManager instance is made. So How about using like this?

```
private int WIDTH;
private int HEIGHT;
```

**Review by dev-moonsc — CHANGES_REQUESTED** [review]
LGTM :)
Commit c5569b9 was previously applied in develop(52cb3fd7) by rabasing.
Please rebase feature/barrier to develop before merging.

File: src/engine/ItemManager.java [hunk 3]
```diff
@@ -252,11 +244,14 @@ private Entry<Integer, Integer> operateBarrier() {
         int middle = WIDTH / 2 - 39;
         int range = 200;
         this.barriers.clear();
-
-        this.barriers.add(new Barrier(middle, HEIGHT - 100));
-        this.barriers.add(new Barrier(middle - range, HEIGHT - 100));
-        this.barriers.add(new Barrier(middle + range, HEIGHT - 100));
-
+        int BarrierY = HEIGHT - 100;
```
Comment (dev-moonsc) [inline src/engine/ItemManager.java:247]: How about change BarrierY little bit lowerside?

```
int BarrierY = HEIGHT - 70
```

If Barriers appear too hight in screen, Overlap issue with the Block or Blocker sprite can be occured

**Comment by bamcasa** [comment]
Refactor BarrierY to 70
remove static

**Pull request merged**